tests/shouldwork/Cores/Xilinx/Floating.hs

```diff
@@ -64,55 +39,59 @@ playSampleRom clk rst content = (done, out)
   cnt = CEP.register clk rst enableGen 0 $ satSucc SatBound <$> cnt
 
 basicBinaryTB
-  :: forall n d
+  :: forall n d x y z
    . ( KnownNat n
      , KnownNat d
+     , Eq z, ShowX z
      , 1 <= n
      )
   => (   Clock XilinxSystem
-      -> DSignal XilinxSystem 0 Float
-      -> DSignal XilinxSystem 0 Float
-      -> DSignal XilinxSystem d Float
+      -> DSignal XilinxSystem 0 x
+      -> DSignal XilinxSystem 0 y
+      -> DSignal XilinxSystem d z
      )
-  -> Vec n (Float, Float, Float)
+  -> z
+  -> Vec n (x, y, z)
   -> Signal XilinxSystem Bool
-basicBinaryTB comp samples = done
+basicBinaryTB comp zDef samples = done
  where
   (inputX, inputY, expectedOutput) = unzip3 samples
   testInputX = fromSignal $ stimuliGenerator clk rst inputX
   testInputY = fromSignal $ stimuliGenerator clk rst inputY
-  expectOutput = outputVerifier' clk rst (repeat @d 0 ++ expectedOutput)
+  expectOutput = outputVerifier' clk rst (repeat @d zDef ++ expectedOutput)
   done =
-      expectOutput . ignoreFor clk rst en (SNat @d) 0
+      expectOutput . ignoreFor clk rst en (SNat @d) zDef
     . toSignal $ comp clk testInputX testInputY
   clk = tbClockGen (not <$> done)
   rst = resetGen
   en = enableGen
 {-# INLINE basicBinaryTB #-}
 
 basicRomTB
-  :: forall d n
+  :: forall d n res
    . ( KnownNat n
      , KnownNat d
+     , Eq res, ShowX res, BitPack res
      , 1 <= n
      )
   => (   Clock XilinxSystem
       -> DSignal XilinxSystem 0 Float
       -> DSignal XilinxSystem 0 Float
-      -> DSignal XilinxSystem d Float
+      -> DSignal XilinxSystem d res
```

Comment:
Is there a reason why it's less generic than `basicBinaryTB`?

Reply:
Is laziness a valid reason? :-)

Reply:
fixed